### DISCUSSION TITLE
How do resource requirements of the Scheduler scale in relation to the task and workers?

### DISCUSSION THREAD

**Discussion by joyofdata** [dask/dask#7515] [Q&A] [answered]
Hello!

Are resource requirements of Scheduler ... 

- ... constant and hence independent of the task and number of workers?
- ... linearly growing with task and/or workers? (F.x. RAM ~ task and CPU ~ workers)

In other words - do I need more CPU/RAM for the Scheduler process if task is larger and distributed to more workers?

Thanks

**Comment by chrisroat** [comment, marked as answer]
In using the distributed scheduler, I have found that:
- I need to allocate a full CPU.  CPU fluctuates a lot during initialization, running, and completion... but often can be near 100%.  Anecdotally (I haven't measured it precisely), the baseline CPU usage seems higher with more workers.
- The RAM usage is low (sub GB) for small to moderate graphs.  But it seems to grow super-linearly for large graphs (10k+ tasks).  For 100k tasks, I need 1GB of RAM and for 200k tasks, I need 3.5GB of RAM.
